### PR TITLE
cleanup: rename fiber/task -> thread

### PR DESCRIPTION
**Pull request by nashif**
We still have many places talking about tasks and threads, replace those
with thread terminology.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: drivers/timer/arcv2_timer0.c [hunk 1]
```diff
@@ -446,7 +446,7 @@ void _timer_idle_exit(void)
 
 	/*
 	 * Ensure the timer will expire at the end of the next tick in case
-	 * the ISR makes any tasks and/or fibers ready to run.
+	 * the ISR makes any tasks and/or threads ready to run.
```
Comment (ruuddw) [inline drivers/timer/arcv2_timer0.c]: "tasks and/or threads" -> please remove tasks part

**Review by ruuddw — COMMENTED** [review]
Looks like you got all 'fibers', there are more 'tasks' (e.g. in arch/arc/core/atomic,S and reset.S, but more places as well) but I think it is ok to leave them.

File: doc/security/security-overview.rst [hunk 2]
```diff
@@ -309,7 +309,7 @@ security violations and limit their impact:
    [PAUL09]_.
 
 -  **Least privilege** describes an access model in which each user,
-   program, thread, and fiber shall have the smallest possible
+   program, thread shall have the smallest possible
```
Comment (dbkinder) [inline doc/security/security-overview.rst]: change to ", and thread"

File: drivers/timer/loapic_timer.c [hunk 3]
```diff
@@ -569,8 +569,8 @@ void _timer_idle_exit(void)
 	 * Either a non-timer interrupt occurred, or we straddled a tick when
 	 * entering tickless idle. It is impossible to determine which occurred
 	 * at this point. Regardless of the cause, ensure that the timer will
-	 * expire at the end of the next tick in case the ISR makes any tasks
-	 * and/or fibers ready to run.
+	 * expire at the end of the next tick in case the ISR makes any threads
+	 * and/or threads ready to run.
```
Comment (dbkinder) [inline drivers/timer/loapic_timer.c]: delete "and/or threads"

File: include/arch/arc/v2/irq.h [hunk 4]
```diff
@@ -74,7 +74,7 @@ extern void _irq_spurious(void *unused);
  * @brief Disable all interrupts on the local CPU
  *
  * This routine disables interrupts.  It can be called from either interrupt,
- * task or fiber level.  This routine returns an architecture-dependent
+ * thread level.  This routine returns an architecture-dependent
```
Comment (dbkinder) [inline include/arch/arc/v2/irq.h]: change to "either interrupt or thread level."

File: include/arch/arc/v2/irq.h [hunk 5]
```diff
@@ -117,7 +117,7 @@ static ALWAYS_INLINE unsigned int _arch_irq_lock(void)
  * is an architecture-dependent lock-out key that is returned by a previous
  * invocation of irq_lock().
  *
- * This routine can be called from either interrupt, task or fiber level.
+ * This routine can be called from either interrupt, thread level.
```
Comment (dbkinder) [inline include/arch/arc/v2/irq.h]: change to "either interrupt or thread level."

File: include/arch/arm/cortex_m/asm_inline_gcc.h [hunk 6]
```diff
@@ -78,7 +78,7 @@ static ALWAYS_INLINE unsigned int find_lsb_set(u32_t op)
  * @brief Disable all interrupts on the CPU
  *
  * This routine disables interrupts.  It can be called from either interrupt,
- * task or fiber level.  This routine returns an architecture-dependent
+ * thread level.  This routine returns an architecture-dependent
```
Comment (dbkinder) [inline include/arch/arm/cortex_m/asm_inline_gcc.h]: change to "either interrupt or thread level."

File: include/arch/arm/cortex_m/asm_inline_gcc.h [hunk 7]
```diff
@@ -150,7 +150,7 @@ static ALWAYS_INLINE unsigned int _arch_irq_lock(void)
  * architecture-dependent lock-out key that is returned by a previous
  * invocation of irq_lock().
  *
- * This routine can be called from either interrupt, task or fiber level.
+ * This routine can be called from either interrupt, thread level.
```
Comment (dbkinder) [inline include/arch/arm/cortex_m/asm_inline_gcc.h]: change to "either interrupt or thread level."

File: include/arch/x86/arch.h [hunk 8]
```diff
@@ -395,7 +395,7 @@ typedef struct nanoIsf {
  * @brief Disable all interrupts on the CPU (inline)
  *
  * This routine disables interrupts.  It can be called from either interrupt,
- * task or fiber level.  This routine returns an architecture-dependent
+ * thread level.  This routine returns an architecture-dependent
```
Comment (dbkinder) [inline include/arch/x86/arch.h]: change to "either interrupt or thread level."

File: include/arch/x86/arch.h [hunk 9]
```diff
@@ -441,7 +441,7 @@ static ALWAYS_INLINE unsigned int _arch_irq_lock(void)
  * is an architecture-dependent lock-out key that is returned by a previous
  * invocation of irq_lock().
  *
- * This routine can be called from either interrupt, task or fiber level.
+ * This routine can be called from either interrupt, thread level.
```
Comment (dbkinder) [inline include/arch/x86/arch.h]: change to "either interrupt or thread level."

File: subsys/logging/kernel_event_logger.c [hunk 10]
```diff
@@ -108,7 +108,7 @@ void _sys_k_event_logger_context_switch(void)
 	 * The mechanism we use to log the kernel events uses a sync semaphore
 	 * to inform that there are available events to be collected. The
 	 * context switch event can be triggered from a task. When we signal a
-	 * semaphore from a task and a fiber is waiting for that semaphore, a
+	 * semaphore from a a thread is waiting for that semaphore, a
```
Comment (dbkinder) [inline subsys/logging/kernel_event_logger.c]: delete the extra a in  "a a thread"

**Review by dbkinder — DISMISSED** [review]
some edits to the changes

**Comment by nashif** [comment]
@dbkinder please check now

File: arch/x86/include/kernel_arch_func.h [hunk 11]
```diff
@@ -45,14 +45,14 @@ static inline void kernel_arch_init(void)
...
+ * @brief Set the return value for the specified thread (inline)
  *
- * @param fiber pointer to fiber
+ * @param thread pointer to thread
  * @param value value to set as return value
  *
  * The register used to store the return value from a function call invocation
- * is set to <value>.  It is assumed that the specified <fiber> is pending, and
- * thus the fibers context is stored in its TCS.
+ * is set to @ value.  It is assumed that the specified @a thread is pending, and
```
Comment (dbkinder) [inline arch/x86/include/kernel_arch_func.h]: ``@a`` on that first ``@``, right?

**Review by dbkinder — CHANGES_REQUESTED** [review]
one question/typo, otherwise all good

**Pull request merged**